http/cookies.lua

```diff
@@ -0,0 +1,325 @@
+local http_patts = require "lpeg_patterns.http"
+local util = require "http.util"
+local psl = require "psl"
+
+local function parse_set_cookie(text_cookie, host, path, time)
+	assert(time, "missing time value for cookie parsing")
+	local key, value, matched_cookie = assert(http_patts.Set_Cookie:match(text_cookie, 1))
+	local cookie = {
+		creation = time;
+		last_access = time;
+		persistent = not not (matched_cookie.expires or matched_cookie["max-age"]);
+		domain = matched_cookie.domain or host;
+		path = matched_cookie.path or path;
+		secure = matched_cookie.secure or false;
+		http_only = matched_cookie.httponly or false;
+		key = key;
+		value = value;
+		host_only = not not matched_cookie.domain;
+		same_site = matched_cookie.same_site;
+	}
+	if matched_cookie["max-age"] then
+		cookie.expires = time + tonumber(matched_cookie["max-age"])
```

Comment:
RFC 6265 section 5.2.2 says it should only be in base10 `DIGIT` with an optional negative, so I'm adding some code to properly handle it that way.